include/swift/AST/DiagnosticsSema.def

```diff
@@ -2805,6 +2805,11 @@ ERROR(assigning_noescape_to_escaping,none,
 ERROR(general_noescape_to_escaping,none,
       "using non-escaping parameter %0 in a context expecting an @escaping closure",
       (Identifier))
+ERROR(converting_noescape_to_generic_param,none,
+      "converting non-escaping value to %0 may allow it to escape",
+      (Identifier))
```

Comment:
if you make this a Type, you can combine the two diagnostics and just pass in the Any type in the second case.

Reply:
Ah interesting, somehow I haven't realized that archetype is also a 'type' :)

Reply:
Done!